starport/pkg/spn/proposal.go

```diff
@@ -40,45 +50,52 @@ type ProposalAddValidator struct {
 }
 
 // ProposalList lists proposals on a chain by status.
-func (c *Client) ProposalList(ctx context.Context, acccountName, chainID string, status ProposalStatus) ([]Proposal, error) {
+func (c *Client) ProposalList(ctx context.Context, acccountName, chainID string, status ProposalStatus, proposalType ProposalType) ([]Proposal, error) {
 	var proposals []Proposal
 	var spnProposals []*genesistypes.Proposal
 
 	queryClient := genesistypes.NewQueryClient(c.clientCtx)
 
+	// Dispatch spn proposal status
+	var spnStatus genesistypes.ProposalStatus
 	switch status {
-	case ProposalPending:
-		res, err := queryClient.ListProposals(ctx, &genesistypes.QueryListProposalsRequest{
-			ChainID: chainID,
-			Status: genesistypes.ProposalStatus_PENDING,
-			Type: genesistypes.ProposalType_ANY_TYPE,
-		})
-		if err != nil {
-			return nil, err
-		}
-		spnProposals = res.Proposals
-	case ProposalApproved:
-		res, err := queryClient.ListProposals(ctx, &genesistypes.QueryListProposalsRequest{
-			ChainID: chainID,
-			Status: genesistypes.ProposalStatus_APPROVED,
-			Type: genesistypes.ProposalType_ANY_TYPE,
-		})
-		if err != nil {
-			return nil, err
-		}
-		spnProposals = res.Proposals
-	case ProposalRejected:
-		res, err := queryClient.ListProposals(ctx, &genesistypes.QueryListProposalsRequest{
-			ChainID: chainID,
-			Status: genesistypes.ProposalStatus_REJECTED,
-			Type: genesistypes.ProposalType_ANY_TYPE,
-		})
-		if err != nil {
-			return nil, err
-		}
-		spnProposals = res.Proposals
+	case ProposalStatusAll:
```

Comment:
```go
var statuses = map[ProposalStatus]genesistypes.ProposalStatus{
      ProposalStatusAll: genesistypes.ProposalStatus_ANY_STATUS,
}
```

Not blocking but would be easier for code readability. (same needed for the other swtich stmt)

Reply:
We have this structure already
```
var toStatus = map[genesistypes.ProposalStatus]ProposalStatus{
	genesistypes.ProposalStatus_PENDING:  ProposalPending,
	genesistypes.ProposalStatus_APPROVED: ProposalApproved,
	genesistypes.ProposalStatus_REJECTED: ProposalRejected,
}
```

We could name the one above `statusFromSPN` and the opposite `statusToSPN`, what do you think?

Reply:
yep :+1: but if that's good, we can complete these after the demo too.

Reply:
Let's do it now 🏃‍♂️